### PR TITLE
chore(dependencies): Update non-breaking deps

### PR DESCRIPTION
**Pull request by AllenBW**
Updates all non-breaking deps, cept storybook cuz thats being done in #269 

SO explaining what you're seeing here: 

* snapshots were updated, cuz patternfly was updated
* a few linty things, cuz ✨ 
* that one nasty `.js` 

[Good point made below, ](https://github.com/patternfly/patternfly-react/pull/293#pullrequestreview-108616242) adding a storybook to this for convenience: https://AllenBW.github.io/patternfly-react/

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -55,43 +55,43 @@
...
+    "jest": "^22.4.3",
+    "jest-cli": "^22.4.3",
+    "node-sass": "^4.8.3",
+    "prettier": "^1.11.1",
     "prettier-eslint": "^8.8.1",
-    "prop-types": "^15.6.0",
+    "prop-types": "^15.6.1",
     "raf": "^3.4.0",
-    "react": "^16.2.0",
+    "react": "^16.3.0",
```
Comment (dmiller9911) [inline package.json]: 🎉

File: package.json [hunk 2]
```diff
@@ -21,12 +21,12 @@
     "breakjs": "^1.0.0",
     "classnames": "^2.2.5",
     "css-element-queries": "^1.0.1",
-    "patternfly": "^3.38.0",
-    "react-bootstrap": "^0.31.5",
+    "patternfly": "^3.42.0",
+    "react-bootstrap": "^0.32.1",
```
Comment (priley86) [inline package.json]: we definitely want to check for regressions on this one... amongst the others ;)

**Comment by AllenBW** [comment]
and yeah, we're suffering from https://github.com/facebook/jest/issues/5772 for a hot second...

File: package.json (hunk 2 above)
Comment (AllenBW) [reply package.json]: @priley86 added storybook in first post 😋

**Comment by priley86** [comment]
did some preliminary looks... this is really nice... 
* storybook loads much much faster, both locally and on your rawgit 👍 
* breadcrumbs are fixed 👍 

maybe we can start to investigate some other React 16.3 improvements in a future PR?

**Comment by priley86** [comment]
interestingly enough... travis build cache was failing something after all these dependency updates in Node 6... but clearing it's cache fixed us ;)

for anyone needing to attempt this in the future in travis ... more options -> caches -> (delete master branch cache)... then restart build...  💚 

i'm guessing this will be sparse though...

**Comment by priley86** [comment]
thanks for the updates here @AllenBW @jeff-phillips-18 !!

**Pull request merged**